yt_dlp/extractor/vidyard.py

```diff
@@ -0,0 +1,256 @@
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    extract_attributes,
+    int_or_none,
+    traverse_obj,
+)
+
+
+class VidyardBaseIE(InfoExtractor):
+
+    def _get_formats_and_subtitles(self, video_source, video_id):
+        video_source = video_source or {}
+        formats, subtitles = [], {}
+        for key, value in video_source.items():
+            if key == 'hls':
+                for video_hls in value:
+                    fmts, subs = self._extract_m3u8_formats_and_subtitles(video_hls.get('url'), video_id, headers={
+                        'referer': 'https://play.vidyard.com/',
+                    })
```

Comment:
1. Should this have `fatal=False`?
2. Should this `referer` be sent for Swearnet?

If not, you could consider scrapping the `common_info` at l.232 below, and the `kwargs` of `_process_video_json()`. Instead, have this in VidyardBaseIE:

```py
    _HEADERS = {}
```
And in VidyardIE (`@property` so that a new `dict` is used each time):
```py
    @property
    def _HEADERS(self):
        return {'referer': 'https://play.vidyard.com/'}
```
And then
```suggestion
                    fmts, subs = self._extract_m3u8_formats_and_subtitles(video_hls.get('url'), video_id, headers=self._HEADERS)
```

Reply:
From a cursory check, it appears that use of `fatal=False` in calls to `_extract_m3u8_formats_and_subtitles` is rather inconsistent across other extractors, so I'm unsure as to what would be the preferred behaviour

Reply:
I don't think sending the referrer for Swearnet will cause a problem as the URL requested is going to be under the cdn.vidyard.com domain, however as it didn't send it previously I guess it shouldn't do now

Reply:
Headers extracted as of b8345dd2076859b85857a82e010d4b793978b041

Reply:
I agree that the use of `fatal=False` is a bit various but I think it's probably good to have it.

A general pattern is that a potential HLS manifest URL is extracted but not validated as a URL, except perhaps by calling `determine_ext()` if the JSON context doesn't imply an HLS manifest URL, and then passed to `_extract_m3u8_formats...()`. Passing `fatal=False` makes that a sort of validation. Also, in many cases other media links might be available, and crashing on the HLS manifest extraction would prevent extracting those.

Reply:
Added `fatal=False` in 2dffeb5692830ed28583944f37baf2c91cd16efb